### PR TITLE
Remove upload temp file also on error

### PR DESCRIPTION
**Pull request by jochenschneider**

### CODE REVIEW

**Comment by heartsucker** [comment]
@jochenschneider I think this might be hard to write a unit test for since we'd have to have a request fail, but did you do a manual test to see that the files got clean by force throwing an exception?

File: core/src/main/scala/org/genivi/sota/core/storage/S3PackageStore.scala [hunk 1]
```diff
@@ -42,28 +42,32 @@ class S3PackageStore(credentials: S3Credentials)
...
     writePackage(packageId, fileData, sink)
+      .andThen { case _ =>
+        log.debug(s"removing temp file ${file.getAbsolutePath}")
+        Try(file.delete())
+      }
   }
 
+  override def store(packageId: PackageId, filename: String, fileData: Source[ByteString, Any]):
+                                                                          Future[(Uri, PackageSize, DigestResult)] =
+    storeAndRemove(packageId, File.createTempFile(filename, ".tmp"), fileData)
```
Comment (koshelev) [inline core/src/main/scala/org/genivi/sota/core/storage/S3PackageStore.scala]: Is `storeAndRemove` used somewhere else or can we "squash" `store` and `storeAndRemove` into single function?

Comment (jochenschneider) [inline core/src/main/scala/org/genivi/sota/core/storage/S3PackageStore.scala]: I factored it out in order to be able to unit test it.

**Comment by jochenschneider** [comment]
@heartsucker i made a unit test for it.

**Pull request merged**